manim/mobject/graphing/functions.py

```diff
@@ -113,7 +113,7 @@ def __init__(
         use_vectorized: bool = False,
         **kwargs,
     ):
-        self.function = function
+        self.function: Callable[[float], Point3D] = lambda t: np.asarray(function(t))
```

Comment:
I would like to discuss this change.

Because we are allowing the user to pass a `Callable` which returns a `Point3DLike`, if I don't touch any of the other code, then `.get_function()` returns a `Callable[[float], Point3DLike]`, `.get_point_from_function(t) returns a `Point3DLike` and, most importantly, `.function(t)` returns a `Point3DLike`, which is not ideal.

`ParametricFunction.function(t)` is called inside `CoordinateSystem.input_to_graph_point(x, graph)`, which is typed to return a `Point3D`. If we don't do any other changes, then this function might return `graph.function(x)` which is a `Point3DLike`, not a `Point3D`.

We have, at least, 3 options here:
- Do what I'm doing here and nest the passed function inside a new one which casts the result to a `Point3D` and returns it. In this way, all the previous calculations I mentioned return `Point3D`. It works, but maybe nesting functions is not ideal.
- Do nothing inside `ParametricFunction` and just let everything return `Point3DLike`. The end user should expect functions to return `Point3DLike` if they passed a callable with that return type. Internal functions which call `.function()` should cast the result to a NumPy array if they want to use it as such (which I don't like a lot). For example, the call to `graph.function(x)` inside `CoordinateSystem.input_to_graph_point(x, graph)` should be wrapped inside a `np.asarray(...)` to make it a `Point3D`.
- Change the parameter type to `Callable[[float], Point3D]`. The user should pass a function with a `Point3D` return type. I don't like that option, because it removes flexibility for the user. If they don't notice that they were expected to return a NumPy array and return something else instead, things might break in unexpected ways for them.

Reply:
Honestly, I like the current solution of a wrapper function. It's simpler and makes it easier to write correct code.
The only thing I might suggest is making it a named function instead of a lambda, so it looks nicer on stack traces?

Reply:
I was also thinking of that! I decided to implement it just now.